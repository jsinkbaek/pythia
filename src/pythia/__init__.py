__all__ = ["periodograms","iterative_prewhitening","resampling","conversions"]

from . import periodograms
from . import iterative_prewhitening
from . import conversions
from . import resampling