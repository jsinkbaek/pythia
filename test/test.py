import numpy as np
import matplotlib.pyplot as plt

from matplotlib import rc

from pythia.periodograms import scargle
from pythia.iterative_prewhitening import run_ipw, run_ipw_v02


# rc('text', usetex=True)
rc('font',**{'family':'serif','serif':['Computer Modern Roman']})
# plt.rcParams['text.latex.preamble']=[r"\usepackage{amsmath}"]
plt.rcParams['xtick.labelsize']=12
plt.rcParams['ytick.labelsize']=12


if __name__=="__main__":

  # times,fluxes = np.loadtxt("./test.data",unpack=True)
  times,fluxes = np.loadtxt("./test_tess.dat",unpack=True)
  nu_w,amp_w = scargle(times,np.ones_like(times),f0=-5.,fn=5.,norm='amplitude')
  nu,amp = scargle(times,fluxes-np.mean(fluxes),fn=50.,norm='amplitude')


  figw,axw = plt.subplots(1,1,figsize=(6.6957,6.6957),num=2)
  axw.plot(nu_w,amp_w,'k-')
  axw.set_xlabel(r'$\nu\,\,{\rm [d^{-1}]}$',fontsize=14)
  axw.set_ylabel(r'${\rm Amplitude\,\,[ppm]}$',fontsize=14)
  axw.set_title(r'${\rm Window}$',fontsize=14)



  fig,ax = plt.subplots(1,1,figsize=(6.6957,6.6957),num=1)
  ax.plot(nu,amp,'k-')
  ax.set_xlabel(r'$\nu\,\,{\rm [d^{-1}]}$',fontsize=14)
  ax.set_ylabel(r'${\rm Amplitude\,\,[ppm]}$',fontsize=14)
  ax.set_title(r'${\rm Example}$',fontsize=14)
  plt.show()

  yerr = 0.0005* np.ones_like(times)
  residuals, model, offsets, \
  frequencies, amplitudes, \
  phases, stop_criteria = run_ipw(times,fluxes-np.mean(fluxes), yerr, maxiter=5, fn=30.)

  # yerr = 0.0005* np.ones_like(times)
  # residuals, outpars= run_ipw_v02(times,fluxes-np.mean(fluxes), yerr, maxiter=5, fn=30.)
  #
  # frequencies = outpars['frequency']
  # amplitudes = outpars['amplitude']
  # phases = outpars['phase']
  # offsets = np.zeros_like(phases)
  # offsets[0] += outpars['offset']
  # stop_criteria = outpars['snr']

  np.savetxt('test.out',np.array([offsets,frequencies,amplitudes,phases]).T)
  fig,ax = plt.subplots(1,1,figsize=(6.6957,6.6957))

  print(' C + A*sin( 2*pi*f*(t-t0)+phi )')
  nu_,amp_ = scargle(times, residuals, fn=6.5, norm='amplitude')

  print(offsets)
  print(frequencies)
  print(amplitudes)
  print(phases)
  print(stop_criteria)

  outstr = '{} -- C: {:.6f} -- A: {:.6f} -- f: {:.6f} -- phi: {:.6f} -- SNR: {:.6f}'
  for ii,freq in enumerate(frequencies):
      print(outstr.format(ii, offsets[ii], amplitudes[ii], freq, phases[ii],stop_criteria[ii]))
      ax.axvline(freq, linestyle='--',color='red',alpha=0.4)
      ax.axhline(amplitudes[ii], linestyle='--',color='red',alpha=0.4)

  ax.plot(nu,amp,'-',color='grey')
  ax.plot(nu_,amp_,'k-')
  ax.set_xlabel(r'$\nu\,\,{\rm [d^{-1}]}$',fontsize=14)
  ax.set_ylabel(r'${\rm Amplitude\,\,[ppm]}$',fontsize=14)
  ax.set_title(r'${\rm Example}$',fontsize=14)
  plt.show()
