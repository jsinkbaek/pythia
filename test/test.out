0.000000000000000000e+00 1.559048717994945044e+01 1.250157065204871238e-03 -9.417820082790787639e-01
0.000000000000000000e+00 2.016409678366206037e+01 1.067550259529726278e-03 -3.118620764963663916e+00
0.000000000000000000e+00 1.678296976201120927e+01 7.555385025632184058e-04 3.104312397224382014e+00
0.000000000000000000e+00 1.428957835921962172e+01 6.384654143987907341e-04 -1.536813720463194199e+00
0.000000000000000000e+00 1.285176025040525971e+01 3.980822120352023991e-04 2.146179931947316710e+00
